lib/regexp/regexp.go

```diff
@@ -0,0 +1,258 @@
+//                                  so for instance $1 represents the text of the first submatch.
+//
+//     re.replace_all(src , replFunc) - Returns a copy of the given string in which all matches of the regular expression re
+//                                      have been replaced by the return value of the replacement function applied to the
+//                                      matched substring. The replacement returned by replacement function is substituted directly,
+//                                      without using Expand.
+//
+//     re.split(src, max) - Returns a tuple of strings between all the matches of the regular expression re.
+//                          If max > 0, at most max strings will be returned knowing that the last string will
+//                          be the unsplit remainder. If max == 0, an empty tuple will be returned. If max < 0,
+//                          all strings will be returned. The parameter max is optional, by default no limit
+//                          will be applied.
+//
+var Module = &starlarkstruct.Module{
+	Name: "regexp",
+	Members: starlark.StringDict{
+		"compile": starlark.NewBuiltin("compile", compile),
+	},
+}
+
+func compile(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		pattern string
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &pattern); err != nil {
+		return starlark.None, err
+	}
+
+	if strings.Contains(pattern, `\C`) {
+		return nil, fmt.Errorf(`The byte-oriented pattern \C is not supported`)
+	}
+
+	re, err := regexp.Compile(pattern)
+	if err != nil {
+		return nil, err
+	}
+	return &Regex{re: re}, nil
+}
+
+func toTuple(strs []string) starlark.Tuple {
+	var vals starlark.Tuple
+	for _, s := range strs {
+		vals = append(vals, starlark.String(s))
+	}
+	return vals
+}
+
+// hashString computes the FNV hash of s.
+func hashString(s string) uint32 {
+	var h uint32
+	for i := 0; i < len(s); i++ {
+		h ^= uint32(s[i])
+		h *= 16777619
+	}
+	return h
+}
+
+// Regex is a starlark representation of a compiled regular expression
+type Regex struct {
+	re *regexp.Regexp
+}
+
+// String implements the Stringer interface
+func (r *Regex) String() string { return r.re.String() }
+
+// Type returns a short string describing the value's type.
+func (r *Regex) Type() string { return "regexp" }
+
+// Freeze renders time immutable. required by starlark.Value interface.
+// The interface regex presents to the starlark runtime renders it immutable,
+// making this a no-op
+func (r *Regex) Freeze() {}
+
+// Hash returns a function of x such that Equals(x, y) => Hash(x) == Hash(y)
+// required by starlark.Value interface
+func (r *Regex) Hash() (uint32, error) { return hashString(r.re.String()), nil }
+
+// Truth returns the truth value of an object required by starlark.Value
+// interface. Any non-empty regexp is considered truthy
```

Comment:
"Non-empty" is ambiguous: do you mean the regular language is empty, or that the pattern string is empty?

I can't see how this feature would be useful. Let's say regexp values are always True (and document this at the top).